port/ztpf/omrosdump.c

```diff
@@ -0,0 +1,343 @@
+#include <stdint.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+#include <errno.h>
+#include <tpf/c_eb0eb.h>
+#include <tpf/c_proc.h>
+#include "j9port.h"
+#include "portpriv.h"
+#include "j9osdump_helpers.h"
+#include "j9signal.h"
+#include <signal.h>
+#include <sys/time.h>
+#include <sys/resource.h>
+#include <sys/stat.h>
+#include <dirent.h>
+#include <libgen.h>
+
+typedef struct iproc IPROC;
+#include "safe_storage.h"
+
+/**
+ * \fn Callback from CCCPSE: preemptible error.
+ *
+ * The z/TPF kernel will call us back at this function when it (the z/TPF
+ * program interrupt handler) determines that the error is 'preemptible'
+ * (i.e. is a 'soft' interrupt which is caused by division by zero or a
+ * null-pointer reference).
+ *
+ */
+static void
+__attribute__((used))
+ztpf_preemptible_err_ep(void)
+{
+   // let the port library do the work.
+   ztpf_preemptible_helper();
+}
+
+/**
+ * \fn Callback from CCCPSE: non-preemptible error.
+ *
+ * The z/TPF kernel will call us back at this function when it (the z/TPF
+ * program interrupt handler) determines that the error is 'non-preemptible'
+ * (i.e. is a 'hard' interrupt) caused by any number of programming errors
+ * considered serious enough to question the state of the JVM.
+ */
+
+static void
+__attribute__((used))
+ztpf_nonpreemptible_err_ep(void)
+{
+   // let the port library do the work.
+   ztpf_nonpreemptible_helper();
+}
+
+
+/**
+ * \fn  Create an ELF core dump file of the user space
+ *
+ *
+ *        This function determines if there is a Java Dump Buffer available pertinent to this
+ *        thread/process. If one is not, we know we were called from the system dump agent's
+ *        executing a hook (such as -Xdump:system:events=vmstop) and arrange for one to be
+ *        created right now. If a dump pertinent to us is available, we use it. We then call
+ *        ztpfBuildCoreDump() in order to build an ELF core dump from its contents, then
+ *        write it to file.
+ *
+ *        After we're done writing the file, zeroize the Java Dump Buffer pointer in the
+ *        DIB block.
+ *
+ * \param[in] portLibrary       The port library.
+ * \param[in,out] filename      Buffer for filename. On input, describes a requested filename.
+ * \param[in] dumpType          Type of dump to perform. Ignored for z/TPF.
+ * \param[in] userData          Implementation specific data. Ignored for z/TPF.
+ * \return 0 on success, non-zero otherwise.
+ *
+ * \note <TT>filename</TT> buffer can not be NULL.
+ * \note User must allocate and free <TT>filename</TT> buffer.
+ * \note <TT>filename</TT> buffer length is platform dependent, assumed to be <TT>EsMaxPath</TT> or <TT>MAX_PATH</TT>
+ * \note If <TT>filename</TT> buffer is empty, an error will be returned. Should it
+ *        contain only a filename (no path), one will be chosen for it.
+ */
+UDATA
+j9dump_create(struct J9PortLibrary *portLibrary, char *filename, char *dumpType, void *userData)

```

Comment:
We seem to be introducing quite a lot of J9 and Java related code in this commit. I'm not sure this belongs at the OMR layer. Can this be abstracted to be used with OMR? And if not shouldn't this code belong in an upstream project consuming OMR?